src/solvers/refinement/README.md

```diff
@@ -6,59 +6,101 @@
 
 \section string_solver_interface String solver interface
 
+The basic role of the solver is to answer whether the set of equations given
+is satisfiable. One example usage, is to determine whether an assertion in a
+program can be violated.
+For instance, CBMC and JBMC, convert a input program and property to check
+about this program to a set of equations. These equations are fed to a solver,
+which is one of the last step in CBMC and JBMC, as it tells us whether the
+property holds or can fail.
+
+The secondary role of the solver is to provide a satisfying assignment of
+the variables of the equations, this can for instance be used to construct
+a trace.
+
+The string solver is particularly aimed at string logic, but since it inherits
+from other solvers it is also capable of handling arithmetic, array logic,
+floating point operations etc.
+
+An example of a problem given to string solver could look like this:
+
+~~~~~
+return_code == cprover_string_concat_func(
```

Comment:
It would be good to explain what these primitives are and where they come from.

Reply:
This is explained in the section "String primitives". Should I add a pointer to the section here?